js/components/dateRange.js

```diff
@@ -0,0 +1,280 @@
+    }
+  },
+  created: function() {
+    Fliplet.Hooks.on('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  destroyed: function() {
+    Fliplet.Hooks.off('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  methods: {
+    initDaterange: function() {
+      var $vm = this;
+
+      if (this.dateRange && !this.$refs.dateRange) {
+        return;
+      }
+
+      if (!this.value.start || !this.value.end && ['default', 'always'].indexOf(this.autofill) > -1 && (this.required || this.autofill === 'always')) {
+        $vm.value = {
+          start: this.today,
+          end: this.today
+        };
+      }
+
+      this.dateRange = Fliplet.UI.DateRange(this.$refs.dateRange, {
+        required: this.required || this.autofill === 'always',
+        forceRequire: false,
+        value: this.value,
+        readonly: this.readonly
+      });
+
+      this.dateRange.change(function(value) {
+        if (!value || !value.start || !value.end) {
+          this.selectedRange = '';
+        }
+
+        $vm.value = value;
+        $vm.updateValue();
+      });
+    },
+    getDate: function(option) {
+      switch (option) {
+        case 'today':
+          return {
+            start: moment().locale('en').format('YYYY-MM-DD'),
+            end: moment().locale('en').format('YYYY-MM-DD')
+          };
+        case 'yesterday':
+          return {
+            start: moment().subtract(1, 'days').locale('en').format('YYYY-MM-DD'),
+            end: moment().subtract(1, 'days').locale('en').format('YYYY-MM-DD')
+          };
+        case 'tomorrow':
+          return {
+            start: moment().add(1, 'days').locale('en').format('YYYY-MM-DD'),
+            end: moment().add(1, 'days').locale('en').format('YYYY-MM-DD')
+          };
+        case 'nextWeek':
+          return {
+            start: moment().add(1, 'days').locale('en').format('YYYY-MM-DD'),
+            end: moment().add(7, 'days').locale('en').format('YYYY-MM-DD')
+          };
+        case 'lastWeek':
+          return {
+            start: moment().subtract(7, 'days').locale('en').format('YYYY-MM-DD'),
+            end: moment().subtract(1, 'days').locale('en').format('YYYY-MM-DD')
+          };
+        case 'nextMonth':
+          return {
+            start: moment().add(1, 'days').locale('en').format('YYYY-MM-DD'),
+            end: moment().add(30, 'days').locale('en').format('YYYY-MM-DD')
+          };
+        case 'lastMonth':
+          return {
+            start: moment().subtract(30, 'days').locale('en').format('YYYY-MM-DD'),
+            end: moment().subtract(1, 'days').locale('en').format('YYYY-MM-DD')
+          };
+        default:
+          break;
+      }
+    },
+    onBeforeSubmit: function(data) {
+      // Empty date fields are validated to null before this hook is called
+      if (this.autofill === 'always' && data[this.name] === null) {
+        data[this.name] = this.defaultSource === 'submission' ?
+          {
```

Comment:
Put the `?` and `:` in the new line

```suggestion
          ? {
```